crates/forge_permission_resolver/src/permissions_cache.rs

```diff
@@ -0,0 +1,261 @@
+use regex::Regex;
+use std::collections::HashMap;
+use std::env;
+use std::path::PathBuf;
+use std::time::Duration;
+
+use crate::permissions_resolver::PermissionHashMap;
+use crate::serde::{SerializablePermissionHashMap, ToStringMap};
+use std::fs::{self, File};
+use std::io::{Read, Write};
+use tracing::debug;
+
+const CACHE_EXPIRATION: Duration = Duration::from_secs(60 * 60 * 24 * 7); // 1 week
+
+fn default_cache_path() -> PathBuf {
+    if let Ok(home) = env::var("HOME") {
+        PathBuf::from(home).join(".cache/fsrt")
+    } else {
+        PathBuf::new()
+    }
+}
+
+/// Struct to hold cache-related settings
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct CacheConfig {
+    use_cache: bool,
```

Comment:
You can just treat `None` as not using a cache path. Also, the ttl variable is unnecessary if it's just going to be a constant.